### PR TITLE
gc: reset $ftotal4/$ftotal6 per continent so data-less continents render 0

### LINKED ISSUE #1507: gc: stale $ftotal4/6 count renders for continents with no data
Deferred from PR #1506 review (CodeRabbit inline finding on `src/usr/local/www/pfblockerng/pfblockerng.php` ~line 1671).

**Finding.** In `pfblockerng_get_countries()`, `$ftotal4`/`$ftotal6` are assigned only inside the per-type parse branch (`${'ftotal'.$type} = count(${'coptions'.$type});`) and are never reset between continent iterations. When a continent has no data for a type (continent file missing or empty `coptions`), the heredoc for that continent renders the **previous** continent's count next to an empty options list. `$options4/$options6` are `unset()` at the continent tail and refilled `''` by the `??=` guard, so the list itself is correctly empty — only the count is stale.

**Why deferred.** This is pre-existing behaviour, byte-for-byte preserved by PR #1506 on purpose: that PR's contract was warning-removal with zero behaviour change beyond the documented defect fixes, and both its independent verifier and adversarial reviewer traced this exact carry and confirmed pre-image parity (see PR #1506 commits `c15cff97`/`5a54790d` and the review audit comment). Changing the rendered count is a separate, user-visible fix.

**Suggested fix.** Reset `$ftotal4 = $ftotal6 = 0;` at the top of each continent iteration (alongside where `$options4/$options6` re-derive), so an empty continent renders `0`. One-line change + a red assertion extending `tests/php/PfbOptionsHeredocMultiContinentTest.php` (continent 2's rendered count must be 0, not continent 1's).

**Issue-gate record.** Producer: supported path — `gc` run when any continent file is missing/empty (fresh install before GeoIP download, or MaxMind data gaps). Privilege: admin CLI/cron. Hand-crafted: no. Impact: cosmetic-to-minor — wrong count rendered into the generated country-code page. Black-box repro: run `php pfblockerng.php gc` with one continent file present and one absent; inspect the generated page's per-continent totals.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

### PR DESCRIPTION
**Pull request by andrebrait**
Fixes #1507.

## What

In `pfblockerng_get_countries()` (the `gc` verb), `$ftotal4`/`$ftotal6` are assigned only inside the per-type `coptions`-guarded build block and were never reset between continent iterations. A continent with no data for a type (missing/empty continent file) therefore rendered the **previous** continent's count (`$options_countries4_cnt`/`$options_countries6_cnt`, the country selects' `size` attribute) next to its — correctly empty — options list. This pre-existing behaviour was traced and deliberately preserved byte-for-byte by PR #1506 (warning-removal contract), then deferred here as a user-visible fix.

The fix resets `$ftotal4 = $ftotal6 = 0;` at the top of each continent iteration, so a data-less continent renders `0`. The per-continent reset makes the #1506 function-top `$ftotal4 = $ftotal6 = 0;` init dead code (the fixed 9-entry `$geoip_files` loop always runs), so that line is removed — a single init site remains (review round 1 finding).

## Red -> green proof (executed)

`tests/php/PfbOptionsHeredocMultiContinentTest.php` gains a fourth positionally-extracted anchor (the continent-loop-top region, so the test proves the reset lands in the right place, not a hand-copy) and a new test capturing the heredoc-equivalent count renders per continent.

RED on untouched production code (test file `git hash-object` = `1ed5098589bf68a1025b68ee56600ea68de702ba`):

```text
1) PfbOptionsHeredocMultiContinentTest::testSecondContinentWithNoDataRendersZeroCounts
issue #1507: $options_countries4_cnt must render 0 on a data-less continent, not the previous continent's stale $ftotal4
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'0'
+'1'
Tests: 2, Assertions: 7, Failures: 1.
```

GREEN after the one-line reset, test file byte-identical (`git hash-object` still `1ed5098589bf68a1025b68ee56600ea68de702ba`):

```text
OK (2 tests, 8 assertions)
```

## Tier-A coverage (test mandate #4)

New module `tests/smoke/ui/test_render_dataless_continent_count.py` (`ui_render`): the seeded MaxMind corpus gives every one of the 9 continents at least one entry, so the stays-empty `$ftotal` path was never driven in-tier. The test manufactures a data-less continent on the live box — Antarctica's cc files stripped to their `# Continent` headers (header-only, not missing, to stay independent of the separate `$continent_en` carry defect tracked in #1515) — reruns `gc`, and asserts the Antarctica page passes the Tier-A oracle with both country selects rendering `size="0"`, while Africa (the continent whose counts a stale carry would have leaked) renders nonzero sizes in the same run. Baseline and post-restore assertions pin before-state and self-encapsulation (restore + regen in `finally`). Authored, not executed in this session (no live VM); the Tier-A CI leg runs it. Tier B is not required: no new page, no multi-step flow, no structural change.

## Gates

`scripts/agent/run-gates.sh --diff origin/devel` on the rebased head:

```text
GATE PASS: python3 -m pytest
GATE PASS: ruff check .
GATE PASS: ruff format --check .
GATE PASS: mypy tests/
GATE PASS: python3 scripts/check_composer_vendor.py
GATE PASS: php -l src/usr/local/www/pfblockerng/pfblockerng.php
GATE PASS: php -l tests/php/PfbOptionsHeredocMultiContinentTest.php
GATE PASS: vendor/bin/phpunit
GATE PASS: composer phpstan
GATE PASS: composer phpcs -- --standard=phpcs.xml.dist src/
GATES: PASS
```

`git diff origin/devel --name-only | python3 scripts/check_coverage_pairing.py` → `Coverage pairing OK: no src<->tests or www<->ui-tests violation.`

Full PHPUnit suite: 3794 tests / 22583 assertions OK (the 1 deprecation + 1 notice are pre-existing `pfblockerng.inc` curl items on PHP 8.5, untouched by this diff).

## Deferred

- #1515 — `$continent`/`$continent_en` stale carry writes a data-less continent's page to the previous continent's path (outside-diff review finding, pre-existing, out of #1507's scope).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Fixed continent pages with no country data incorrectly displaying counts carried over from another continent.
  - Empty IPv4 and IPv6 country lists now correctly display counts of `0`.
  - Prevented warnings when rendering continents without available country data.
  - Preserved correct counts for continents that contain data.

### CODE REVIEW

File: tests/smoke/ui/test_render_dataless_continent_count.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+carry.
+
+Fail-before / pass-after: pre-fix, the mutated ``gc`` run regenerates the
+Antarctica page with Africa's (the previous continent's) nonzero counts as the
+select sizes, so the ``size="0"`` assertions fail; they pass only with the
+per-continent reset. The off-appliance red run is executed and pinned by
+``tests/php/PfbOptionsHeredocMultiContinentTest.php``.
+
+AUTHORED, NOT EXECUTED this session (no live VM) -- run via
+``pytest tests/smoke/ui -m ui_render --override-ini="addopts="`` on the fan-out VM.
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply tests/smoke/ui/test_render_dataless_continent_count.py:28]: Skipping this one — the narration checker does not apply here. `scripts/check_comment_narration.py` scans only `src/` and `scripts/` (`_SCAN_ROOTS = ("src", "scripts")`), which is deliberate policy, not an oversight — `.agents/policy/coding.md` states the rule is "enforced on added lines under `src/` + `scripts/`". This comment is in `tests/`, out of scope by design.

Verified rather than assumed: ran the checker against this PR's diff — `python3 scripts/check_comment_narration.py --diff origin/devel` → exit 0 (matches the green "Comment narration" CI check). None of the checker's patterns match the flagged phrases regardless. Left as-is per the unenforced-rule triage in the landing policy.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

**Comment by andrebrait** [comment]
## Review audit — PR #1511

**Adversarial review:** one independent read-only pass. Model: small tier (Sonnet). Profile: `full`, effort `xhigh` (a `www/` behaviour change is present). Size: 3 files, +218/-10.

**Verdict: clean — no blocking findings.**

- **Contract:** the reset `$ftotal4 = $ftotal6 = 0;` sits at the top of the single outer `foreach ($geoip_files ...)` loop, so it covers all nine continents uniformly and resets both v4 and v6 symmetrically — structurally impossible to reset only one sibling. Runs before the inner per-type loop and before any `continue`; no first/last-continent edge case. Dead function-top init correctly removed.
- **Test honesty (mutation-proven):** PHPUnit run four ways — baseline green; full revert → the count assertion goes red; reset **only v4** → the IPv6 (`countries6`) assertion catches it; reset **only v6** → the v4 assertion catches it. The vacuous-IPv6-assertion fix from commit `5fbaf2a4` genuinely binds, symmetrically. `check_coverage_pairing.py` on the diff → OK.
- **Smoke test** (`test_render_dataless_continent_count.py`, Tier-A `ui_render`) provides the mandated `www/` coverage. It is authored-not-executed this session (the UI fan-out CI leg is a label-gated opt-in and the PR carries no `ui-tests` label); non-vacuity was proven statically — the `size` attribute it asserts on is the raw unfiltered `$ftotal6` string, the same code path the executed PHP-level v4-only mutation already flips — and devel's daily UI backstop exercises it live.

**CodeRabbit:** one inline thread (narration nitpick, Major) — triaged MOOT/SKIP: the narration checker scans only `src`+`scripts` per `coding.md`, and running it against the diff returns exit 0 (matches the green CI gate). Replied on the thread.

🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.
